fe/src/main/java/org/apache/doris/qe/ShowExecutor.java

```diff
@@ -1215,12 +1215,17 @@ private void handleShowTablet() throws AnalysisException {
                 }
                 boolean stop = false;
                 Collection<Partition> partitions = new ArrayList<Partition>();
-                List<String> partitionNames = showStmt.getPartitionNames();
                 if (showStmt.hasPartition()) {
-                    for (Partition partition : olapTable.getPartitions()) {
-                        if (partitionNames.contains(partition.getName())) {
-                            partitions.add(partition);
+                    List<String> partitionNames = showStmt.getPartitionNames();
+                    for (String partName : partitionNames) {
+                        Partition partition = olapTable.getPartition(partName);
+                        if (partition == null) {
```

Comment:
I will reconsider this when implementing loading process of temp partition.
If I change the origin `getPartition` method, it may change lots of codes in this PR.

Reply:
OK